clients/gitlabrepo/commits.go

```diff
@@ -107,15 +76,42 @@ func (handler *commitsHandler) setup() error {
 					Message:       commit.Message,
 					SHA:           commit.ID,
 				})
+				continue
 			}
 
-			// Casting the Reviewers into clients.Review.
-			var reviews []clients.Review
+			// Two GitLab APIs for reviews (reviews vs. approvals)
+			// Use a map to consolidate results from both APIs by the user ID who performed the
+			reviews := make(map[int]clients.Review)
 			for _, reviewer := range mergeRequest.Reviewers {
-				reviews = append(reviews, clients.Review{
-					Author: &clients.User{ID: int64(reviewer.ID)},
-					State:  "",
-				})
+				if reviewer.State == "unreviewed" {
+					continue
+				}
+				reviews[reviewer.ID] = clients.Review{
+					Author: &clients.User{Login: reviewer.Username, ID: int64(reviewer.ID)},
+					State:  "COMMENTED",
+				}
+			}
+
+			approvals, _, err := handler.glClient.MergeRequests.GetMergeRequestApprovals(
```

Comment:
Can we avoid this extra API call by using GitLab's GraphQL API - https://docs.gitlab.com/ee/api/graphql/reference/#mergerequest ?